python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -324,17 +324,18 @@ class RandomForestClassifier(Base):
         cdef size_t params_t64
         if self.n_cols:
             # only if model has been fit previously
-            model_pbuf_bytes = self._get_protobuf_bytes()
+            self._get_protobuf_bytes()  # Ensure we have this cached
             params_t = <uintptr_t> self.rf_forest
-            rf_forest = \
-                <RandomForestMetaData[float, int]*>params_t
+            if params_t:
+                rf_forest = \
+                    <RandomForestMetaData[float, int]*>params_t
+                state["rf_params"] = rf_forest.rf_params
+
             params_t64 = <uintptr_t> self.rf_forest64
-            rf_forest64 = \
-                <RandomForestMetaData[double, int]*>params_t64
-            state["rf_params"] = rf_forest.rf_params
-            state["rf_params64"] = rf_forest64.rf_params
-        else:
```

Comment:
curious about why the else option to set `model_pbuf_bytes = bytearray()` is not required anymore